### PR TITLE
Add Capacity Planner CLI

### PR DESCRIPTION
**Pull request by shunjikawabata**
Add Capacity Planner CLI, which is a stand-alone tool to extract peak resource usage values and corresponding timestamps for a given GCP project, time range and timezone.

### CODE REVIEW

File: tools/capacity-planner-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+Options:
+  --project_id TEXT               GCP project ID where the Cloud Monitoring
+                                  API is called against.  [required]
+  --end_time [%Y-%m-%dT%H:%M:%S%z]
+                                  The end time in ISO 8601 format of the time
+                                  interval for which           results should
+                                  be returned. Default is now.           e.g.
+                                  2022-10-03T05:23:02+09:00
+  --duration_minutes INTEGER      The number of minutes in the time interval
```
Comment (aymanfarhat) [inline tools/capacity-planner-cli/README.md:27]: Are there any limits on the duration? e.g. maximum, would be good to specify if any

File: tools/capacity-planner-cli/README.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+## Example
+
+The following command will produce a CSV file in `logs/result.csv`.
+
+```
+python capacity_planner.py --project_id my-awesome-project --end_time '2022-10-15T18:30:00+09:00' --duration_minutes 60
+```
+
+
+| product_name<br>NaN    | metric_name<br>NaN            | metrics<br>resource.region | <br>unit | <br>resource.location | value<br>NaN | time<br>NaN               |
```
Comment (aymanfarhat) [inline tools/capacity-planner-cli/README.md:46]: When running this on my end, I noticed that the first row always contains NaN values, any idea why I would be getting that? e.g.
```
python capacity_planner.py --project_id gsk-workflows-pipeline-demo --duration_minutes=10000

```

```
,product_name,metric_name,metrics,metrics,value,time
,nan,nan,resource.location,unit,nan,nan
0,Cloud Storage,QPS,eu,1/s,0.019999777774074013,2023-01-26 12:20:36+00:00
1,Cloud Storage,QPS,us,1/s,0.04999944443518503,2023-01-26 10:20:36+00:00
0,Cloud Storage,Egress MiB/s,eu,MiBy/s,0.15225682833681153,2023-01-26 12:57:36+00:00
1,Cloud Storage,Egress MiB/s,us,MiBy/s,0.020324938955993504,2023-01-26 10:20:36+00:00
```

**Review by aymanfarhat — CHANGES_REQUESTED** [review]
Looks good overall, added some points that would be good to check out before merging, but nothing major. Very nice tool! :)

File: tools/capacity-planner-cli/queries.toml [hunk 3]
```diff
@@ -0,0 +1,191 @@
+# queries.toml
```
[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/queries.toml:1]: Ack

File: tools/capacity-planner-cli/capacity_planner.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    def _clean_query(self, query: str) -> str:
+        """Remove newlines and multiple whitespaces."""
+        q = ''.join(query.splitlines())
+        q = ' '.join(q.split())
+
+        return q
+
+    def _extract_index_names(self, time_series_pager) -> List[str]:
+        descriptors = time_series_pager._response.time_series_descriptor
+        index_names = [label_descriptor.key for label_descriptor in descriptors.label_descriptors]  # noqa: E501
```
[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/capacity_planner.py]: Yes, my original intention was omitting flake8 E501. I've changed my mind and made it follow PEP-8.

File: tools/capacity-planner-cli/capacity_planner.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+import toml
+
+from typing import Any, Generator, List, Tuple
+
+import datetime
+
+
+class CapacityPlanner(object):
+    def __init__(self, project_id: str) -> None:
+        # Cloud Monitoring client
```
[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/capacity_planner.py]: Ack

File: tools/capacity-planner-cli/capacity_planner.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+    # Set the timezone for the results from `end_time`
+    tz = end_time.tzinfo
+
+    df = client.query_as_dataframe(
+        tz=tz, end_time=end_time, duration_minutes=duration_minutes)
+
+    print(df)
+
+    df.to_csv(output)
```
[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/capacity_planner.py]: Changed to use logging. Thanks

[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/capacity_planner.py]: Changed to include 'logs' folder in the repo.

File: tools/capacity-planner-cli/capacity_planner.py [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+    if end_time is None:
+        end_time = datetime.datetime.now(datetime.timezone.utc)
+
+    # Set the timezone for the results from `end_time`
+    tz = end_time.tzinfo
+
+    df = client.query_as_dataframe(
+        tz=tz, end_time=end_time, duration_minutes=duration_minutes)
+
+    print(df)
```
[the remark this answers is not in the document]
Comment (shunjikawabata) [reply tools/capacity-planner-cli/capacity_planner.py]: Changed to use logging instead.

File: tools/capacity-planner-cli/README.md (hunk 1 above)
Comment (shunjikawabata) [reply tools/capacity-planner-cli/README.md:27]: Added help comments to indicate the maximum

File: tools/capacity-planner-cli/README.md (hunk 2 above)
Comment (shunjikawabata) [reply tools/capacity-planner-cli/README.md:46]: Changed to exclude the index column. Thanks, it is really a good catch.

**Comment by shunjikawabata** [comment]
@aymanfarhat I've addressed your comments. Please take another look. Thanks for your valuable feedback :)

**Comment by aymanfarhat** [comment]
@shunjikawabata Thanks! LGTM.

**Review by aymanfarhat — APPROVED** [review]

**Pull request merged**